representation/transparency_using_selection/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import { createRootViewer } from "./common/init";
+import { setStructureTransparency } from "molstar/lib/mol-plugin-state/helpers/structure-transparency";
+import { Queries, QueryContext, StructureProperties, StructureSelection } from "molstar/lib/mol-model/structure";
+
+async function init() {
+    // Create viewer
+    const plugin = await createRootViewer();
+    
+    // Download PDB
+    const fileData = await plugin.builders.data.download(
+        { url: "https://models.rcsb.org/4hhb.bcif", isBinary: true }
+    );
+
+    // Load PDB and create representation
+    const trajectory = await plugin.builders.structure.parseTrajectory(fileData, "mmcif");
+    const data = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
+
+    // Get Structure and StructureRef objects from data
+    const struct = data?.structure.cell?.obj?.data!;
+    const structRef = plugin.managers.structure.hierarchy.current.structures.find((sr)=>sr.cell.obj?.data === struct)
+    
+    // Query all chains with ID A or B
+    const ctx = new QueryContext(struct)
+    const query = Queries.generators.chains({
+        chainTest: l => StructureProperties.chain.label_asym_id(l.element) == 'A' || StructureProperties.chain.label_asym_id(l.element)== 'B'
```

Comment:
`l` is a QueryContext object. In fact, it is a direct reference to `ctx` (I did a console.log(ctx === l) to check). My understanding is that the callback is executed as we pop through the stack of chains. And the `element` it points to, changes after each iteration.
In my tests, it is called 14 times in this example, so probably once for each entity for each chain (proteins + heme + hoh in chains A, B, C, D and 2 phosphate groups)

The line has a repetition which might make it harder to read.
I'd suggest:
```ts
chainTest: queryCtx => {
            const chainName = StructureProperties.chain.label_asym_id(queryCtx.element);
            return chainName === 'A' || chainName === 'B';
        }
```

Reply:
I agree this is easier to read